pubsub/pubsub.go

```diff
@@ -532,6 +534,20 @@ func (s *Subscription) Receive(ctx context.Context) (_ *Message, err error) {
 			m := s.q[0]
 			s.q = s.q[1:]
 			s.throughputCount++
+			if s.ackFunc == nil {
+				// Add a finalizer that complains if the Message we return isn't
+				// acked or nacked.
+				_, file, lineno, ok := runtime.Caller(1)
+				runtime.SetFinalizer(m, func(m *Message) {
+					if !m.isAcked {
+						var caller string
+						if ok {
+							caller = fmt.Sprintf(" (%s:%d)", file, lineno)
+						}
+						log.Printf("A pubsub.Message was never Acked or Nacked%s", caller)
```

Comment:
I suspect we'll get complaints about this, and we'll end up turning it into a metric. But we can cross that bridge later.

Reply:
Ack.